crates/sncast/src/starknet_commands/account/add.rs

```diff
@@ -52,6 +53,18 @@ pub struct Add {
     #[allow(clippy::struct_field_names)]
     #[clap(long)]
     pub add_profile: Option<String>,
+
+    #[clap(flatten)]
+    pub rpc_args: RpcArgs,
+}
+
+impl Provider for Add {
```

Comment:
I don't think it is worth to add additional layer here. I would just do `add.rpc_args.get_provider(&config).await?`